activationapi/pom.xml

```diff
@@ -34,7 +34,7 @@
             jakarta.activation
         </activation.extensionName>
         <activation.moduleName>
-            jakarta.activation
+            jakarta.activation.api
```

Comment:
Why do you think the module name needs to change?

This purposely has the same module name as the full API+implementation module so that you can
compile against this Maven artifact with a dependency on the jakarta.activation module and then
run with the jakarta.activation API+implementation module.

If you compiled against a jakarta.activation.api module, there would be nothing to run against.

Reply:
> Why do you think the module name needs to change?

I'm sorry I can't offer a better solution; my suggestion is based on the perception that the approach of reusing the same module name died with Java 9 modules.

> This purposely has the same module name as the full API+implementation module so that you can compile against this Maven artifact with a dependency on the jakarta.activation module and then run with the jakarta.activation API+implementation module.

I think Java modules neither work this way, nor are they supposed to.

In the classpath-world the JVM doesn't care if both the api jar and the impl(+api) jar end up on the classpath.

In the module-world the JVM simply won't start if it finds two modules with the same module declaration.

This is a problem, because pretty much every non-trivial project will end up depending on both the api and the impl(api) jar via random transitive dependencies.

Reply:
> > This purposely has the same module name as the full API+implementation module so that you can compile against this Maven artifact with a dependency on the jakarta.activation module and then run with the jakarta.activation API+implementation module.
> 
> I think Java modules neither work this way, nor are they supposed to:
> 
> In the classpath-world the JVM doesn't care if both the api jar and the impl(+api) jar end up on the classpath.

The fact that it "happens to work" doesn't mean "you were supposed to do it".
It's still a mistake to include both.

> In the module-world the JVM simply won't start if it finds two modules with the same module declaration.
> 
> This is a problem, because pretty much every non-trivial project will end up depending on both the api and the impl(api) jar via random transitive dependencies.

Only if people use the wrong dependencies.

Let's fix the actual mistakes people have made rather than break an intended use
case to accommodate these mistakes.

> > If you compiled against a jakarta.activation.api module, there would be nothing to run against.
> 
> This might make sense in a Maven module world, but sadly doesn't translate to the Java module world, as far as I can see.

I don't see why you say that.  The .api module is incomplete, you can't run against it.
There's no other module that adds the missing pieces.  And even if there were you
wouldn't want your standard API module to depend on one specific implementation
module from one vendor.

Activation was not designed with the common API/implementation split that you'll find
in many other Java EE APIs.  Trying to force it into that model just isn't going to work.